java-checks-test-sources/src/main/java/checks/UnusedPrivateMethod.java

```diff
@@ -11,10 +11,15 @@
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
+import javax.enterprise.event.Observes;
 
 import static java.util.Collections.emptySet;
 
 class UnusedPrivateMethodCheck {
+  
+  private void initNc(@AnotherAnnotation Object object) {} // Noncompliant
+  private void init(@Observes Object object) {} //Compliant, javax.enterprise.event.Observes is an exception to the rule
+  private void init(@Observes Object object, String test) {}
```

Comment:
I would add one with the fully qualified annotation and one with a custom annotation: 
```java
private void foo(@javax.enterprise.event.Observes Object object) {} // Compliant

// custom annotation
@interface @Observes {}

private void bar(@checks.UnusedPrivateMethodCheck.Observes Object object) {} // Noncompliant - will be a FN in absence of semantic
```